### ISSUE TITLE
Safari, iframes, different ports, lost databases

### ISSUE THREAD

**Issue by getflourish** [dfahlander/Dexie.js#1413]
TLDR; I’m aware that Safari might be the issue here but I want to see if this is a known issue or could be related to Dexie:

I've encountered a strange bug with IndexedDB that seems to be related to iframes.

I have a website running on `http://localhost:1337`
That website embeds an **iframe** that points to `http://localhost:3000`

**The iframe is saving to IndexedDB.** The embedding page is not accessing IndexedDB at all.

**The problem:**
Safari shows (3!) databases: one for :3000 and two for :1337
When I restart Safari, only the one database for :3000 remains. All data is lost.

**Expected:**
One database for :3000 with persisted content.

While using the app, multiple databases exist. Only one has the actual data (port :3000)

<img width="433" alt="Bildschirmfoto 2021-10-17 um 18 24 27" src="https://user-images.githubusercontent.com/183989/137636596-10edb81c-dc19-45e8-9d35-ae395f676dea.png">

After restarting Safari: all data is gone. The remaining database is for the wrong port.

<img width="831" alt="Bildschirmfoto 2021-10-17 um 18 25 07" src="https://user-images.githubusercontent.com/183989/137636605-61a32d53-3856-42fc-9ff9-fad8b9bf4300.png">
